### ISSUE TITLE
torch median / nanmedian w/ nans speed

### ISSUE THREAD

**Issue by Ivan-Dimitrov** [pytorch/pytorch#63870]
Hello, I am trying to calculate the median for a NxM array (N ~ 100 and M ~ 25,000). The normal median function is fast enough, but I need to mask out some values for each row. I use nanmedian; however, I see it is a magnitude slower (.5 ms vs .05 ms).

I am running the following code 
```
# x is NxM
x[x == 0] = float("nan")
# x_medians is N
x_medians = torch.nanmedian(x, dim=1)
```
If I comment out, `x[x == 0] = float("nan")`, I can recover the faster performance but of course that's the wrong answer.

Is there a way to achieve what I want while keeping performance. I am on pytorch 1.8.1 btw.

**Comment by cpuhrsch** [comment]
@Ivan-Dimitrov - could you expand on your use case a bit more? You say that you need to mask out some values for each row. Do you mean you want to them be ignored akin to https://github.com/pytorch/rfcs/pull/27?

**Comment by Ivan-Dimitrov** [comment]
After reading, I believe my request is captured by the linked rfc @cpuhrsch. To make sure we are on the same page, here is an example
```
x = torch.tensor(
[5,  6, 10, 12]
[12, 3,  2, 7]
[9,   8,  1, 2]
)

mask = torch.tensor(
[True, True, True, False]
[False, True, False, True]
[True, True, False, False]
)
```


```
res_medians = torch.maskedMedian(x, mask)
print(res_medians)
>> [6, 3, 8]  #note torch.median returns the lower of two elements if the length of the array is even
```

Basically, torch.nanmedian has this functionality but it's slower than it should be. 

Let me know if there is more clarification I should provide.
